subprojects/core/src/main/java/org/gradle/internal/progress/HasBuildOperationContext.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.progress;
+
+public interface HasBuildOperationContext {
+
+    /**
+     * Returns a human-consumable display name for this domain object.
+     */
+    String getBuildOperationDisplayName();
```

Comment:
The problem is that `Project.getDisplayName()` already exists. What it returns, is not what we use for the build operation name anymore. That was changed in https://github.com/gradle/gradle/commit/a71cf6226dbcbc4a54e0fbcff34497d9cbf95dac.


So I needed a second description (and a method with another name). It probably makes sense that we want things to be described a bit differently in the context of build operations. `HasBuildOperationContext` (or however we call that) can also be enriched with more methods ind the future to get more context information.